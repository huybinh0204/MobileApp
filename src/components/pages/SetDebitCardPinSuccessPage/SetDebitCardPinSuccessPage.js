import { useNavigation, useRoute } from '@react-navigation/native';
import React from 'react';
import { IconSvg, MainButton } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS } from '_constants';
import strings from '_localization';
import { normalize } from '_utilities/screen';
import { SuccesPageContainer, SuccesPageTitle } from './SetDebitCardPinSuccessPage.styles';

const SetDebitCardPinSuccessPage = () => {
  const { params } = useRoute();
  const { popToTop } = useNavigation();

  const { isChangePinVariant } = params;

  return (
    <SecondaryScreenLayout testID="SetDebitCardPinSuccessPage" showBackButton={false}>
      <SuccesPageContainer>
        <IconSvg
          width={normalize(250)}
          height={normalize(250)}
          icon={ICONS.cardSuccessActivation}
        />
        <SuccesPageTitle>
          {isChangePinVariant ? strings.card.changePinSuccess : strings.card.setInitialPinSuccess}
        </SuccesPageTitle>
        <MainButton accessibilityLabel="Go To Card Tab" onPress={popToTop}>
          {strings.linkCard.awesome}
        </MainButton>
      </SuccesPageContainer>
    </SecondaryScreenLayout>
  );
};

export default SetDebitCardPinSuccessPage;
