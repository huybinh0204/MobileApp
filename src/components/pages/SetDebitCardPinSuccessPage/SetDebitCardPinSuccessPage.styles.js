import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const SuccesPageContainer = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  padding: theme.spacing.m,
}));

export const SuccesPageTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  marginTop: theme.spacing.m,
  marginBottom: 'auto',
  paddingHorizontal: theme.spacing.m,
  textAlign: 'center',
}));
