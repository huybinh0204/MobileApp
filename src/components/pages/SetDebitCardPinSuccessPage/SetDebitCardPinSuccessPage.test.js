import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import SetDebitCardPinSuccessPage from './SetDebitCardPinSuccessPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('SetDebitCardPinSuccessPage', () => {
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useNavigation.mockReturnValue(mockedNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', async () => {
    const { toJSON } = render(<ComponentWithProviders component={SetDebitCardPinSuccessPage} />);

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should navigate to the card tab main screen', async () => {
    const { getByAccessibilityLabel } = render(
      <ComponentWithProviders component={SetDebitCardPinSuccessPage} />
    );

    const goToTop = getByAccessibilityLabel('Go To Card Tab');

    await fireEvent.press(goToTop);

    expect(mockedNavigation.popToTop).toBeCalledTimes(1);
  });
});
