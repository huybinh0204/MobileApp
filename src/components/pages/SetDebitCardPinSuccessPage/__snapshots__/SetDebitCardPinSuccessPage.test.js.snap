// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetDebitCardPinSuccessPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SetDebitCardPinSuccessPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={453}
                              width={453}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 43.5,
                                    "lineHeight": 60,
                                    "marginBottom": "auto",
                                    "marginTop": 58,
                                    "paddingHorizontal": 58,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Your card has been activated.
                            </Text>
                            <View
                              accessibilityLabel="Go To Card Tab"
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#3D3397",
                                  "borderRadius": 10,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Awesome!
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
